mlflow/cli.py

```diff
@@ -554,7 +575,50 @@ def gc(older_than, backend_store_uri, run_ids):
     else:
         run_ids = run_ids.split(",")
 
-    for run_id in run_ids:
+    if not skip_experiments:
+        deleted_experiment_ids = []
+        deleted_older_experiment_ids = []
+        next_experiment_page_token = None
+        time_threshold = get_current_time_millis() - time_delta
+        while True:
+            page_results = backend_store.search_experiments(
+                view_type=ViewType.DELETED_ONLY,
+                page_token=next_experiment_page_token,
+            )
+            for experiment in page_results:
+                if older_than is None:
+                    deleted_experiment_ids.append(experiment.experiment_id)
+                elif (
+                    experiment.last_update_time is not None
+                    and experiment.last_update_time <= time_threshold
+                ):
```

Comment:
We've merged https://github.com/mlflow/mlflow/pull/6979. We can use filter experiments by `last_update_time` like this:

```
backend_store.search_experiments(
    view_type=ViewType.DELETED_ONLY,
    filter_string=f"last_update_time < {time_threshold}"
    page_token=next_experiment_page_token,
)
```